azurerm/resource_arm_application_gateway_test.go

```diff
@@ -260,6 +260,32 @@ func TestAccAzureRMApplicationGateway_probesPickHostNameFromBackendHTTPSettings(
 	})
 }
 
+func TestAccAzureRMApplicationGateway_backendHttpSettingsHostName(t *testing.T) {
```

Comment:
Could we get another test with 3 steps:
 - set pick true
 - set hostname
 - unset hostname

Reply:
I have a doubt regarding the test. If I set pick true and then I try to set hostname it will throw an error. Until here it's fine, i can check for that error. But I dont understand the following step. I think that unsetting hostname won't do anything because of the previous error.

Should I add a test just to check for that error (both pick and host_name set)?

Reply:
@katbyte I've added the test TestAccAzureRMApplicationGateway_backendHttpSettingsHostNameAndPick checking for the error and it's passing.
Please let me know if you think there is something else to check for.